File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.js
function solution(s, skip, index) {
    var answer = '';
    alphabet = new Set(['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n',
                    'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'])
    newAlpha = Array.from(alphabet).filter(char => !skip.includes(char))
    newAlpha.sort()
    
    for (idx = 0; idx < s.length; idx++) {
        newIdx = newAlpha.indexOf(s[idx]) + index
        newIdx = newIdx % newAlpha.length
        answer += newAlpha[newIdx]
    }    
    return answer;
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    answer = ''
    alphabet = set(['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n',
                    'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'])
    newAlpha = list(alphabet - set(skip))
    newAlpha.sort()
    
    for idx in range(len(s)):
        alphaIdx = newAlpha.index(s[idx]) + index
        newIdx = alphaIdx % len(newAlpha)
        answer+=newAlpha[newIdx]
    return answer
